addons/project/models/project_project.py

```diff
@@ -49,19 +49,20 @@ def _compute_attached_docs_count(self):
             project.doc_count = docs_count.get(project.id, 0)
 
     def _compute_task_count(self):
-        task_count_per_project = {
-            project.id: count
-            for project, count in self.env['project.task'].with_context(
-                active_test=any(project.active for project in self)
-            )._read_group(
-                [('state', 'not in', list(CLOSED_STATES)), ('project_id', 'in', self.ids)],
-                ['project_id'],
-                ['__count'],
-            )
-        }
+        project_and_state_counts = self.env['project.task'].with_context(
+            active_test=any(project.active for project in self)
+        )._read_group(
+            [('project_id', 'in', self.ids)],
+            ['project_id', 'state'],
+            ['__count'],
+        )
+        task_counts_per_project_id = defaultdict(lambda: [0, 0])
+        for project, state, count in project_and_state_counts:
+            task_counts_per_project_id[project.id][state in CLOSED_STATES] += count
```

Comment:
This feels a bit hackish, I think it would be more readable with a dict instead of a list